src/components/AutomaticRevenueCenter.tsx

```diff
@@ -129,13 +129,36 @@ export const AutomaticRevenueCenter = () => {
 
   const setupDefaultOrderServiceType = useRef(false)
 
+  // "pathname !== cateringPath" added for catering wizard on 2021-08-02
+  // catering wizard resets the revenue center when a service type is selected
   useEffect(() => {
-    if (!revenueCenter && loadingRevenueCenter === 'idle' && !resetToOLO) {
+    if (
+      !revenueCenter &&
+      loadingRevenueCenter === 'idle' &&
+      !resetToOLO &&
+      pathname !== cateringPath
+    ) {
       dispatch(fetchRevenueCenter(defaultRevenueCenterId))
       dispatch(setLocationType('fallbackRevenueCenter'))
       setupDefaultOrderServiceType.current = true
     }
-  }, [dispatch, loadingRevenueCenter, revenueCenter, resetToOLO])
+  }, [dispatch, loadingRevenueCenter, revenueCenter, resetToOLO, pathname])
+
+  useEffect(() => {
+    if (!orderType && revenueCenter) {
+      // TODO: this assumes that we'll always have an order type when arriving
+      // at /catering-menu, so we can safely reset to OLO in this case
+      // however, this will not be the case if we add a "descriptive" catering
+      // menu that users can get to WITHOUT an orderType
+      if (revenueCenter.revenue_center_type === 'CATERING') {
+        setResetToOLO(true)
```

Comment:
This will always take it to OLO. Generally speaking, the 2 cases in the useEffect seem a bit like dirty fixes. If we fix the resetOrder in the CheckoutPayment page, we could probably reduce this to setResetToOLO to true (though, hopefully, this will never run)